vumi_message_store/memory_backend_manager.py

```diff
@@ -178,6 +182,18 @@ def store_object(self, fake_object):
             new_indexes.sort()
             bucket["indexes"][index_name] = new_indexes
 
+    def _filter_old_indexes(self, fake_object, global_indexes):
+        """
+        Remove the object's key from indexes it no longer specifies
+        """
+        filtered_indexes = {}
+        for index_name, index_data in global_indexes.iteritems():
+            if index_name not in fake_object._current_data["indexes"]:
+                index_data = list(filter(
+                    lambda (_, key): key != fake_object.key, index_data))
```

Comment:
Do we need to keep old entries for indexes that the new object has (potentially different) values for?